src/services/SearchResults.h

```diff
@@ -118,14 +118,15 @@ class SearchResults
 
         unsigned char* GetBytes() { return (nSize > sizeof(m_vBytes)) ? m_pBytes : &m_vBytes[0]; }
         const unsigned char* GetBytes() const { return (nSize > sizeof(m_vBytes)) ? m_pBytes : &m_vBytes[0]; }
-
+        auto GetByte(std::size_t nIndex) noexcept { return m_vBytes[nIndex]; }
+        auto GetByte(std::size_t nIndex) const noexcept { return m_vBytes[nIndex]; }
```

Comment:
This only works in `m_nSize` is 8 or less. See `GetBytes` implementation on the previous line. This should be:
```
unsigned char GetByte(size_t nIndex) { return GetBytes()[nIndex]; }
```
Which is more-or-less exactly what the code was doing before. It was just storing the result of `GetBytes()` in a local variable.

Reply:
sure.